src/storage-client/src/types/connections.rs

```diff
@@ -349,13 +376,59 @@ impl KafkaConnection {
         }
 
         let mut context = BrokerRewritingClientContext::new(context);
+        // TODO(guswynn): factor this out
```

Comment:
Er, factor out to where? This mostly LGTM for the long term! Just the SSH tunnel establishment?